### DISCUSSION TITLE
Getting different form of json output from trivy image scanner

### DISCUSSION THREAD

**Discussion by Shivam-nagar23** [aquasecurity/trivy#5754] [Bugs]
### Description
Getting different format of json file format when scanning image through trivy CLI, Can someone from the team highlight what are the cases this is possible and is this the expected behaviour. I am using AWS Iam role.
{
  "SchemaVersion": 2,
  "ArtifactName": "",
  "ArtifactType": "",
  "Metadata": {
    "OS": {
      "Family": "",
      "Name": "",
      "EOSL": true
    },
    "ImageID": "",
    "DiffIDs": [],
    "RepoTags": [],
    "RepoDigests": [],
    "ImageConfig": 
  },
  "Results": []
}
the second format : 
[
  {
    "Target": "",
    "Type": "alpine",
    "Vulnerabilities": []
  },
  {
    "Target": "",
    "Type": "jar"
  },
  {
    "Target": "",
    "Type": "jar",
    "Vulnerabilities": []
  }
]

### Desired Behavior

It should reflect a particular format json , i am further doing operations on the json through templates, it bugs out as we have different format.

### Actual Behavior

Different format of json data is not desirable behaviour.

### Reproduction Steps

```bash
1.Scan a image through trivy CLI (with -f json -o trivy.json) flags.
2.Gives different format of data with differnt images.
...
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Debug Output

```bash
(export AWS_ACCESS_KEY_ID= '' AWS_SECRET_ACCESS_KEY='' AWS_DEFAULT_REGION=''; trivy image -f json -o /tmp/132/3/2_out.json --timeout 10m image)
```

### Operating System

macOs

### Version

```bash
0.41.0
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @Shivam-nagar23 
Thanks for your report!

Are you having same problem with latest version (v0.48.0)?
If yes, can you share two test images with different JSON formats for research?

> the second format :

It looks like your `second format` is `Result` array. Do you mean you don't get `SchemaVersion` and other fields?
